# HLA-B27 Molecular Mimicry Study â€” MD Simulation & Analysis

 **## ğŸ“˜ Project Overview**
 
This repository documents a complete computational workflow to identify **Klebsiella pneumoniaeâ€“derived peptides** that mimic a **human Annexin peptide (ANX)** in binding to **HLA-B27:05**, a Class-I MHC associated with autoimmune diseases such as **Ankylosing Spondylitis**.

The objective is to investigate whether microbial peptides can:

â€¢	Bind HLA-B27 in a similar way as human peptides.
â€¢	Form stable structural interactions.
â€¢	Exhibit comparable post-MD behavior  

â†’ Supporting a potential **molecular mimicry mechanism**.

I focused on learning:
- Peptide generation and filtering
- HLA-B27 binding predictions
- Sequence similarity and mimicry search
- Peptide docking and molecular dynamics (MD)
- Trajectory analysis (RMSD, RMSF, SASA, PCA, MMGBSA
  
---

 **ğŸ› ï¸## Overview of the Pipeline**
   
1. Generate peptides from human Annexin (anx) and full *K. pneumoniae* (kp)proteome  
2. HLA-B27 binding prediction using NetMHCpan  
3. Sequence similarity & mimic scoring  
4. Docking using AlphaFold-Multimer  
5. Structure optimization with AmberTools23 (tleap)  
6. Molecular Dynamics (MD) simulation in GROMACS 2024.
7. Post-process trajectories (centering, frame skipping, fitting)  
8. Post-MD analysis: RMSD, RMSF, Rg, SASA, H-bonds, MM-GBSA  
9. Identification of strongest microbial mimic

    âš ï¸ **Note: ** The workflow was developed through trial and error, is **not fully automated**, and may **not be fully reproducible**. Scripts are shared to illustrate my learning process.
This workflow includes:

---

 **âš™ï¸ ## Repository Structure**

hlab27-mimicry-study/

â”‚â”€â”€ README.md

â”‚â”€â”€ .gitignore

â”‚â”€â”€ .gitattributes

â”‚â”€â”€ .gitignore

â”‚â”€â”€ LICENSE

â”‚â”€â”€ environment.yml

â”‚

â”œâ”€â”€ data/

â”‚ â””â”€â”€ Human_pep_fasta/

â”‚ â””â”€â”€ kleb_proteome/

â”‚ â””â”€â”€ processed/

â”‚ â””â”€â”€ README.md

â”‚

â”œâ”€â”€ docking/

â”‚ â””â”€â”€  docking_anx_pep/

â”‚ â””â”€â”€ docking_kleb_pep/

â”‚ â””â”€â”€ README.md

â”‚

â”œâ”€â”€ md_simulations/

â”‚ â””â”€â”€  tleap_input/

â”‚ â””â”€â”€  amber_to_gromacs/

â”‚ â””â”€â”€  gromacs_inputs/

â”‚ â””â”€â”€ README.md

â”‚

â”œâ”€â”€ results/

â”‚ â””â”€â”€ plots/

â”‚ â””â”€â”€ summary_table/

â”‚ â””â”€â”€ README.md

â”‚

â”œâ”€â”€ scripts/

â”‚ â””â”€â”€ pep_slicing/

â”‚ â””â”€â”€ pep_mimicking/

â”‚ â””â”€â”€ scripts_md/ 

â”‚ â””â”€â”€ README.md


---

 **ğŸ› ï¸ ## Materials and Methods**

  **### Protein and Peptides:**

o	**Target protein:** 
   
   HLA-B27:05 (Reference PDB structure with similar binding groove, used as model template)  
   
o	**Peptides studied:**

â€¢	Human: ANX
â€¢	Microbial: KP1, KP2, KP3

  **### Software / Tools:**
   
â€¢	Python 3.10+ / Biopython â€“ peptide generation and sequence alignment
â€¢	NetMHCpan â€“ HLA-B27 binding predictions
â€¢	BLAST / Bio.Align â€“ sequence similarity search
â€¢	AlphaFold / HADDOCK â€“ peptide docking
â€¢	AmberTools23 â€“ preparation of peptideâ€“HLA complexes
â€¢	GROMACS 2024 â€“ MD simulations and trajectory analysis
â€¢	gmx_MMPBSA â€“ binding free energy calculations

  **###  Peptide Generation**
   
- Peptides (9â€“12-mers) were generated from Annexin (anx) and the full K. pneumoniae (kp) proteome.
- One set of peptides were generated without any rule HLA-B27 anchor rules on the other hand another set were generated by applying HLA anchor rule:
  **Peptide length: 9â€“10 amino acids
  P1: small residue
  P2: positively charged (Arg, Lys)
  C-terminal (PÎ©): hydrophobic/aromatic (Leu, Phe, Tyr, Met, Val, Trp, Ile)**
- Tools: Biopython  
- Scripts: `generate_anx_peps_with_rule.py`, `generate_kp_peps_without_rule.py` (`/scripts/pep_slicing/`)

 	###  Binding Prediction â€” NetMHCpan
  
- All peptides of anx and kp screened for HLA-B27 binding
- Strong binder (SB) and weak binder (WB) extracted (only 10 strong binders peps from anx and around 90 sb peps from kp were selected)

 	###  Mimicry Scoring
  
- Sequence similarity using Biopython (BLOSUM62)
- Scripts (`/scripts/pep_mimicing/`):
          `check_mimicry_kp_anxn.py` (got around 3 or more kp peptides for each anx peptides)
           `top5_mimicry_kp_anxn.py` (only top 5 anx peptides along with it 28 mimic kp candidates were selected for modeling  from all 10 anx peptide)
  
 	###  Docking â€” AlphaFold-Multimer
  
- ANX + all selected kp models docked with HLA-B27
- Output: `.cif` model + confidence `.json`  
- Visual inspection done in PyMOL (binding orientation and position in HLA groove)
- 3 kp peptides KP1, KP2 and KP3 were selected against one anx pep according to docking score and visual inspection.
- therefore the selected set (total 4)
                        Human pep: anx
                        microbial pep: kp1, kp2, kp3

 	### Structure Optimization & MD Setup
  
- Minimization + solvation via AmberTools23 (tleap)
- Converted to GROMACS using ParmEd
          script: `ParmEd_Amb2Gmx.py`  (`/scripts/script_md/`)

 	### MD Simulation
  
- Software: **GROMACS 2024**
- Water: TIP3P
- Ensemble: NVT â†’ NPT â†’ 1 Âµs production
- Conditions: 300 K, 1 bar, PME electrostatics, 2 fs timestep

md script: `md.sh`  (`/scripts/script_md/`)
Trajectory correction: `md_correction.sh`  (`/scripts/script_md/`)

** 	### Post-MD Analysis**
   
Performed using GROMACS:
- RMSD
- RMSF
- Radius of gyration
- SASA
- Hydrogen bonds
- MM-GBSA free energy (gmx_MMPBSA)
  
analysis script : post_md_analysis.sh (`/scripts/script_md/`)

Plots & summary tables stored in `/results/`.

---

**ğŸ“Š ## Key Findings**

- **ANX (human peptide)** formed the most stable and compact complex with HLA-B27.
- **KP1** showed the closest behavior to ANX in:
  ğŸ§© RMSD  
  ğŸ§© Rg  
  ğŸ§© SASA  
  ğŸ§© Hydrogen bonds  
  ğŸ§© Binding energy
  
  â†’ **we can suggest it as mimic candidate** but further investigation or experimental validation will be required.
- **KP2** showed high flexibility and weak binding,
- **KP3** behaved intermediately
-  kp2 and kp3 can not be suggested as mimic candidate

---

 **ğŸ“˜ ## Reproducibility Note**
   
project was created for learning purposes. The scripts were developed through trial and error and are not fully automated, so the workflow may not be fully reproducible.The scripts are shared mainly to illustrate the approach and my learning process, not as a polished, production-ready workflow. 

---

ğŸ“Œ**ğŸ“‘ ## Citation**
 
If you use this repository, analysis pipeline, or scripts, please cite:
Singh S. (2025). GitHub Repository for HLA-B27 molecular mimicry MD analysis.
Link: https://github.com/singh-sanju/HLA-B27_Peptide_Molecular_Mimicry_MD_Analysis

---

**ğŸ™ ## Acknowledgments**

This work was completed under the mentorship of
Dr Nikolina Å oÅ¡tariÄ‡, Bionanoscience department, Delft University of Technology (TU Delft), Netherlands. I sincerely thank my PI for guidance, supervision, and the opportunity to pursue this research project.
Computational work, including molecular dynamics production runs and post-processing, was performed on the
DelftBlue Supercomputer at TU Delft, using GROMACS 2024 (module load GROMACS/2024r1-openmpi).
I would also like to thank colleagues and lab members at TU Delft for technical help, feedback, and support during this study.


